cli/cmd/component.go

```diff
@@ -158,16 +160,40 @@ func (c *cliState) LoadComponents() {
 					Short:                 component.Description,
 					Annotations:           map[string]string{"type": "component"},
 					Version:               ver.String(),
+					SilenceUsage:          true,
 					DisableFlagParsing:    true,
 					DisableFlagsInUseLine: true,
 					RunE: func(cmd *cobra.Command, args []string) error {
-						cli.Log.Debugw("running component", "component", cmd.Use, "args", args)
+						globalFlags := []*pflag.Flag{}
+						cmd.Flags().VisitAll(func(f *pflag.Flag) {
+							// At runtime, we visit all global flags defined at the root command
+							// and we essentially make a slice of the flag names with their shorthand
+							// so that we can remove them all from the raw arguments (`args`)
+							globalFlags = append(globalFlags, f)
+						})
+
+						// Use the list of global CLI flags to filter the provided arguments and return
+						// the filteres arguments to pass to the underlying component and the real list
+						// of CLI flags. The later is used to parse and then run the global CLI init func
+						filteredArgs, filteredCLIFlags := filterCLIFlagsFromComponentArgs(args, globalFlags)
+
+						// Parse all global CLI flags provided (filtered) by the user, then run the global
+						// CLI init function to initialize our logger, api client, and other global config
+						err := cmd.Flags().Parse(filteredCLIFlags)
+						initConfig() // @afiune NOTE we purposely run this func first and then check the err
+						if err != nil {
+							cli.Log.Debugw("unable to parse global flags",
+								"provided_flags", filteredCLIFlags, "error", err)
+						}
+
+						cli.Log.Debugw("running component", "component", cmd.Use,
+							"args", filteredArgs, "cli_flags", filteredCLIFlags)
 						f, ok := cli.LwComponents.GetComponent(cmd.Use)
 						if ok {
 							// @afiune what if the component needs other env variables
 							envs := []string{fmt.Sprintf("LW_COMPONENT_NAME=%s", cmd.Use)}
 							envs = append(envs, c.envs()...)
-							return f.RunAndOutput(args, envs...)
+							return f.RunAndOutput(filteredArgs, envs...)
```

Comment:
@afiune , How does a component go about respecting some of these global args (i.e. `--json`)?  Should the component specify (i.e. via specs.json) whether or not args should be filtered?

Reply:
I think components should load these options from environmental variables.

Reply:
@afiune , I think this makes sense.  Just to clarify I'm not seeing the CLI consistently set global flags as environment variables.  For instance, on `0.44.1` when I specify:

```
lacework remediate alert 1233214 --nocolor --debug --nocache
```

My environment looks like:
```
[HOSTNAME=f3829371e1fe LW_SUBACCOUNT= LW_API_TOKEN=_007b8a090893066043c34dd85778023e PWD=/ HOME=/root TERM=xterm SHLVL=1 LW_ACCOUNT=dev7.dev7.corp PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin _=/usr/local/bin/lacework NO_COLOR=true LW_COMPONENT_NAME=remediate LW_API_KEY= LW_API_SECRET=]
```

I get `NO_COLOR=true`, but no environment flags around `DEBUG` or `NO_CACHE`.

Reply:
yup, good catch!